content/docs/guides/multi-target-exporter.md

```diff
@@ -0,0 +1,492 @@
+# HELP probe_http_duration_seconds Duration of http request by phase, summed over all redirects
+# TYPE probe_http_duration_seconds gauge
+probe_http_duration_seconds{phase="connect"} 0.062076202999999996
+probe_http_duration_seconds{phase="processing"} 0.23481845699999998
+probe_http_duration_seconds{phase="resolve"} 0.029594103
+probe_http_duration_seconds{phase="tls"} 0.163420078
+probe_http_duration_seconds{phase="transfer"} 0.002243199
+# HELP probe_http_redirects The number of redirects
+# TYPE probe_http_redirects gauge
+probe_http_redirects 1
+# HELP probe_http_ssl Indicates if SSL was used for the final redirect
+# TYPE probe_http_ssl gauge
+probe_http_ssl 1
+# HELP probe_http_status_code Response HTTP status code
+# TYPE probe_http_status_code gauge
+probe_http_status_code 200
+# HELP probe_http_uncompressed_body_length Length of uncompressed response body
+# TYPE probe_http_uncompressed_body_length gauge
+probe_http_uncompressed_body_length 14516
+# HELP probe_http_version Returns the version of HTTP of the probe response
+# TYPE probe_http_version gauge
+probe_http_version 1.1
+# HELP probe_ip_protocol Specifies whether probe ip protocol is IP4 or IP6
+# TYPE probe_ip_protocol gauge
+probe_ip_protocol 4
+# HELP probe_ssl_earliest_cert_expiry Returns earliest SSL cert expiry in unixtime
+# TYPE probe_ssl_earliest_cert_expiry gauge
+probe_ssl_earliest_cert_expiry 1.581897599e+09
+# HELP probe_success Displays whether or not the probe was a success
+# TYPE probe_success gauge
+probe_success 1
+# HELP probe_tls_version_info Contains the TLS version used
+# TYPE probe_tls_version_info gauge
+probe_tls_version_info{version="TLS 1.3"} 1
+```
+
+You can see that the probe was successful and get many useful metrics, like latency by phase, status code, ssl status or certificate expiry in [Unix time](https://en.wikipedia.org/wiki/Unix_time).  
+The blackbox exporter also offers a tiny web interface at [localhost:9115](http://localhost:9115) for you to check out the last few probes, the loaded config and debug information. It even offers a direct link to probe `prometheus.io`. Handy if you are wondering why something does not work.
+
+## Querying multi-target exporters with Prometheus
+
+So far, so good. Congratulate yourself. The blackbox exporter works and you can manually tell it to query a remote target. You are almost there. Now you need to tell Prometheus to do the queries for us.  
+
+Below you find a minimal prometheus config. It is telling Prometheus to scrape the exporter itself as we did [before](#query-exporter) using `curl 'localhost:9115/metrics'`:
+
+NOTE: If you use Docker for Mac or Docker for Windows, you can’t use `localhost:9115` in the last line, but must use `host.docker.internal:9115`. This has to do with the virtual machines used to implement Docker on those operating systems. You should not use this in production.
+
+`prometheus.yml` for Linux:
+
+```yaml
+scrape_configs:
+
+- job_name: blackbox-exporter-metrics
```

Comment:
This is a bit verbose, and metrics is redundant here. Just blackbox will do.

Reply:
I get why you say that, but this is to clarify that it is about the metrics of the exporter itself.
It is aimed at people who want to learn about the concept and named like that for pedagogical reasons.

Reply:
It's redundant though, and I consider it an anti-pattern. This will be blindly copy&pasted, so we should follow best practices.

Reply:
How about `blackbox-exporter-itself`?

Reply:
That's still a bit redundant, I'd personally use blackbox, but could live with blackbox_exporter

Reply:
Redundancy is a feature in this context.

Reply:
It's a bug, as it's extra typing/reading. We don't need every job name ending with prometheus-telemetry-metrics-job (I've seen much worse in practice). Job names should be succinct.

Reply:
Yes, in a normal context it is absolutely a bug. I don’t disagree with you. And I share your frustration about redundant job names in production systems.
But this is an exercise to help people form their mind. We are trying to solve a different problem than usually. Everything is only true within a context and when it changes things can become tricky.
Maybe we can find a solution that satisfies both requirements at once? Something pedagogical and not to verbose.

We have two jobs:
* One about the metrics of the exporter itself. That must be clear to people who are new.
* One about the targets we tell the exporter to test for us. That must also be clear for new people and not implying magic. 

At the same time we want people not to form the bad habit of really long redundant job names.

How about we use a comment? It is a bit worse for understanding and less elegant, but the risk that people adopt bad habits is much lower.

```yaml
 - job_name: blackbox-exporter # to get metrics about the exporter itself 
[…]
 - job_name: blackbox # to get metrics from targets we want the exporter to query for us
```

Reply:
I think you underestimate how we just all copy and past examples from the internet. I am on Brian's side on this one.

Reply:
Good point. I’ll go with the comments solution.

Reply:
We can use comments reasonably liberally, the type of probe should be in the job name though as you can't put all your different probe types in one job.